docs/developer/style.md

```diff
@@ -2,34 +2,348 @@
+- [Program comments](#program-comments)
+- [Documentation](#documentation)
+- [Internal packages](#internal-packages)
+- [Dependency management and Build](#dependency-management-and-build)
+- [Test](#test)
+  - [Table-Driven-Test](#table-driven-test)
+  - [The steps to create a Table-Driven-Test](#the-steps-to-create-a-table-driven-test)
 
 ## Introduction
 
-## Style
+This guideline includes the coding style for all Vald contributors and reviewers. Everyone should follow this guideline to keep the style consistent so everyone can understand and contribute to Vald easier once they learn this guideline. You should have the basic knowledge of how to write Golang before contributing to Vald. If you found any bug please create [a GitHub issue](https://github.com/vdaas/vald/issues/new?assignees=&labels=type%2Fbug%2C+priority%2Fmedium%2C+team%2Fcore&template=bug_report.md&title=) and we will work on it.
+
+Please also read the [Contribution guideline](../../CONTRIBUTING.md) before you start contributing to Vald.
+
+## Code Formatting and Naming Convension
+
+Code formatting and naming conventions affect coding readability and maintainability. Every developer has a different coding style, luckily Golang provides tools to format source code and checking for the potential issue in the source code. We suggest using [gofmt](https://golang.org/cmd/gofmt/) to format the source code in Vald, and [golint](https://github.com/golang/lint). We suggest everyone install the plugin for your editor to automatically format the code once you edit the code.
+But having tools to format source code doesn't mean you do not need to care the formatting of the code, for example:
+
+```go
+// bad
+badStr := "apiVersion: v1\n" +
+   "kind: Service\n" +
+   "metadata:\n" +
+   "  name: grafana\n" +
+   "spec:\n" +
+   "  ports:\n" +
+   "    - name: grafana\n" +
+   "      port: 3000\n" +
+   "      targetPort: 3000\n" +
+   "      protocol: TCP\n"
+
+// good
+goodStr := `apiVersion: v1
+kind: Service
+metadata:
+  name: grafana
+spec:
+  ports:
+    - name: grafana
+      port: 3000
+      targetPort: 3000
+      protocol: TCP
+`
+```
+
+### Project Layout
+
+The project layout includes the folder and the file structure in the project. We follow the [project-layout](https://github.com/golang-standards/project-layout) example in Vald.
+
+### Packages
+
+The package defines the context of the objects in the package, for example, the corresponding methods and structs belong to the corresponding package. Unlike other languages like Java, in Golang we use the package name to declare which context of the object we are going to use. For example in [time](https://golang.org/pkg/time/) package, it defines all the objects about time like `time.Now()` method to get the current time.
+
+Here is the naming conventions of the package:
+
+- All lower case.
+
+```go
+// bad
+package ioUtil
+
+// good
+package ioutil
+```
+
+- No plurals.
+
+```go
+// bad
+package bytes
+
+// good
+package byte
```

Comment:
In golang std there are bytes package, and this example seems not correct `package byte` make problem if we use byte type.